### PR TITLE
warnnotprint

### PR DESCRIPTION
**Pull request by SimonGreenhill**
- work on tests
- API refactored
- fix ci
- upgrade deps
- use logger rather than print

### CODE REVIEW

**Comment by SimonGreenhill** [comment]
Not sure why `gh pr create --fill` decided to add the above details, but this PR just uses log.warn for the check command rather than print

**Comment by xrotwang** [comment]
I think it re-based your changes on HEAD and added the messages of the intermediate commits.

File: src/phlorest/commands/check.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """
 Checks datasets for compliance
 """
+import sys
```
Comment (xrotwang) [inline src/phlorest/commands/check.py:4]: Made it back in, but isn't used.

**Review by xrotwang — APPROVED** [review]

File: src/phlorest/commands/check.py (hunk 1 above)
Comment (SimonGreenhill) [reply src/phlorest/commands/check.py:4]: ahh, I'd used sys.exit at some point. I'll remove this in main now.

**Pull request merged**